### PR TITLE
Add bicep support

### PR DESCRIPTION
**Pull request by chrisghill**
Add the ability to scrape a bicep template and generate a JSON schema from declared bicep params.

Note:  The resulting JSON schema doesn't represent a valid bicep params.json file since bicep requires a custom format where each top level parameter has its value listed under a `value` key. To illustrate, normal JSON would look like:
```json
{
  "foo": "bar"
}
```
Whereas bicep wants the JSON to look like:
```json
{
  "foo": {
    "value": "bar"
  }
}
```

Instead of exposing this peculiarity to the users in our params schema, we just handle the transformation in the provisioner image.

### CODE REVIEW

**Review by xpositivityx — APPROVED** [review]

File: cmd/bicep.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	return bicepCmd
+}
+
+func runBicepInput(cmd *cobra.Command, args []string) error {
+	schema, err := bicep.BicepToSchema(args[0])
+	if err != nil {
+		return err
+	}
+	fmt.Println(schema)
+	return nil
```
Comment (mclacore) [inline cmd/bicep.go:37]: How come this is returning `nil`?

File: pkg/bicep/biceptoschema.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	Metadata    map[string]interface{} `json:"metadata"`
+}
+
+func BicepToSchema(valuesPath string) (string, error) {
+	// using the github.com/Checkmarx/kics parser since he already did the heavy lifting to parse a bicep template
+	parser := bp.Parser{}
+
+	params := new(schema.Schema)
+	params.Type = "object"
+	params.Properties = orderedmap.New[string, *schema.Schema]()
```
Comment (mclacore) [inline pkg/bicep/biceptoschema.go:38]: assuming this is just using the above library to create an ordered map as opposed to unordered?

File: pkg/bicep/biceptoschema.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+		return parseBoolParam(sch, bicepParam)
+	case "string":
+		return parseStringParam(sch, bicepParam, false)
+	case "secureString":
+		return parseStringParam(sch, bicepParam, true)
+	case "array":
+		return parseArrayParam(sch, bicepParam)
+	case "object", "secureObject":
+		return parseObjectParam(sch, bicepParam)
+	default:
```
Comment (mclacore) [inline pkg/bicep/biceptoschema.go:96]: are types like numbers, floats not supported?

File: cmd/bicep.go (hunk 1 above)
Comment (chrisghill) [reply cmd/bicep.go:37]: We need to return something for the `error`, and `nil` represents "no error occurred".

File: pkg/bicep/biceptoschema.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	MaxLength     *uint64            `json:"maxLength"`
+	MaxValue      *uint64            `json:"maxValue"`
+}
+
+type bicepParamMetadata struct {
+	Description string                 `json:"description"`
+	Metadata    map[string]interface{} `json:"metadata"`
+}
+
+func BicepToSchema(valuesPath string) (string, error) {
```
Comment (mclacore) [inline pkg/bicep/biceptoschema.go:32]: rename `valuesPath`?

File: pkg/bicep/biceptoschema.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+		slices.Sort(sch.Required)
+	}
+
+	return nil
+}
+
+func parseArrayType(sch *schema.Schema, value []interface{}) error {
+	items := new(schema.Schema)
+
+	elem := value[0]
```
Comment (mclacore) [inline pkg/bicep/biceptoschema.go:207]: should `value` error be handled here too?

File: pkg/bicep/biceptoschema.go (hunk 2 above)
Comment (chrisghill) [reply pkg/bicep/biceptoschema.go:38]: Yeah, we could just use a `map[string]interface{}` but that is inherently unordered. JSON is technically unordered, so it doesn't matter, but for organization and presentation it makes it a bit nicer.

File: pkg/bicep/biceptoschema.go (hunk 3 above)
Comment (chrisghill) [reply pkg/bicep/biceptoschema.go:96]: Yeah, bicep only support integers currently. https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/data-types#integers

> Floating point, decimal or binary formats aren't currently supported.

File: pkg/bicep/biceptoschema.go (hunk 5 above)
Comment (chrisghill) [reply pkg/bicep/biceptoschema.go:207]: Good catch, I'll add protection for an empty array.

**Review by mclacore — APPROVED** [review]
Nice!

**Pull request merged**